### DISCUSSION TITLE
Filtering in "Database Navigator" is very slow for databases with many tables

### DISCUSSION THREAD

**Discussion by tral** [dbeaver/dbeaver#12176] [Q&A] [answered]
#### System information: 
- MacOS Big Sur 11.2.2
- 21.0.0
- No additional extensions

#### Connection specification:
- MySQL 5.6
- Driver name MySQL 5
- No tunnels or proxies

#### Describe the problem you're observing:
For databases with big number of tables filtering takes several seconds (7-10) and hangs/freezes the application for several seconds. Sometimes after the initial 1st time filtering it starts working instantly, but later it again takes several seconds, it's very annoying.

#### Steps to reproduce, if exist:
Try to connect several databases with 500+ tables and filter table names in Database Navigator section several times. Eventually even if your Mac is powerful app will be frozen for 5-10 seconds.

**Comment by kseniiaguzeeva** [comment]
Thank you for the report. Please try to disable **show statistic info** if it is enabled. It can help.
![Screenshot 2021-03-02 at 11 39 34](https://user-images.githubusercontent.com/51405061/109621537-24cbd080-7b4c-11eb-9bae-872f90cd722b.png)

**Comment by tral** [comment]
Thanks @kseniiaguzeeva it helped a bit, but I still see random freezes, for 3-5 seconds for now (which is not ideal, but is still an improvement).

**Comment by uslss** [comment, marked as answer]
You could try to use [Filter ](https://github.com/dbeaver/dbeaver/wiki/Filter-Database-Objects)item on the context menu of Tables folder
